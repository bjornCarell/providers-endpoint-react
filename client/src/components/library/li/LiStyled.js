import styled from 'styled-components';

export const Li = styled.li`
  display: ${props => (props.inherit ? 'inherit' : 'block')};
  align-self: ${props => {
    if (props.aligSelf) return props.aligSelf;
    if (props.alignSelfEnd) return 'flex-end';
    if (props.alignSelfCenter) return 'center';
    return 'flex-start';
  }};
  flex: ${props => {
    if (props.flexNone) return 0;
    if (props.flexTwo) return 2;
    if (props.flexThree) return 3;
    return 1;
  }};
  flex-direction: ${props => (props.column ? 'column' : 'row')};
  padding: ${props =>
    props.noPadding || props.noPaddingBottom ? 0 : '1.5rem'};
`;
