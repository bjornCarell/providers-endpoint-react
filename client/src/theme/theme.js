export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
