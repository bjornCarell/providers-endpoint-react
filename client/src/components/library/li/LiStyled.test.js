import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { theme } from '../../../theme/theme';
import 'jest-styled-components';
import { Li } from './LiStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Li theme={theme} />, div);
});

it('renders with default conditional styles', () => {
  const tree = renderer.create(<Li theme={theme}>Providers</Li>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'block');
  expect(tree).toHaveStyleRule('align-self', 'flex-start');
  expect(tree).toHaveStyleRule('flex', '1');
  expect(tree).toHaveStyleRule('flex-direction', 'row');
  expect(tree).toHaveStyleRule('padding', '1.5rem');
});

it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Li theme={theme} inherit alignSelfEnd flexTwo column noPadding>
        Providers
      </Li>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'inherit');
  expect(tree).toHaveStyleRule('align-self', 'flex-end');
  expect(tree).toHaveStyleRule('flex', '2');
  expect(tree).toHaveStyleRule('flex-direction', 'column');
  expect(tree).toHaveStyleRule('padding', '0');
});

it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Li theme={theme} alignSelfCenter flexThree>
        Providers
      </Li>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-self', 'center');
  expect(tree).toHaveStyleRule('flex', '3');
});
