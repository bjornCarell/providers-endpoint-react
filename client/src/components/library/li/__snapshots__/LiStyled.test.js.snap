// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with default conditional styles 1`] = `
.c0 {
  display: block;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 1.5rem;
}

<li
  className="c0"
>
  Providers
</li>
`;

exports[`renders with set conditional styles 1`] = `
.c0 {
  display: inherit;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
}

<li
  className="c0"
>
  Providers
</li>
`;

exports[`renders with set conditional styles 2`] = `
.c0 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 1.5rem;
}

<li
  className="c0"
>
  Providers
</li>
`;
